### PR TITLE
Fix toc.yaml for V2 reference docs

### PR DESCRIPTION
**Pull request by taeold**
toc.ts is a short script to generate `toc.yaml` based on the generated markdown file. The output looks as follows

```
toc:
  - title: firebase-functions
    path: /docs/functions/beta/reference/firebase-functions.md
    section: []
  - title: firebase-functions/alerts
    path: /docs/functions/beta/reference/firebase-functions.alerts.md
    section:
      - title: firebase-functions/alerts/alertevent
        path: /docs/functions/beta/reference/firebase-functions.alerts.alertevent.md
        section: []
      - title: firebase-functions/alerts/appdistribution
        path: >-
          /docs/functions/beta/reference/firebase-functions.alerts.appdistribution.md
        section:
          - title: firebase-functions/alerts/appdistribution/appdistributionevent
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.appdistribution.appdistributionevent.md
            section: []
          - title: firebase-functions/alerts/appdistribution/appdistributionoptions
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.appdistribution.appdistributionoptions.md
            section: []
          - title: firebase-functions/alerts/appdistribution/newtesterdevicepayload
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.appdistribution.newtesterdevicepayload.md
            section: []
      - title: firebase-functions/alerts/billing
        path: /docs/functions/beta/reference/firebase-functions.alerts.billing.md
        section:
          - title: firebase-functions/alerts/billing/billingevent
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.billing.billingevent.md
            section: []
          - title: firebase-functions/alerts/billing/planautomatedupdatepayload
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.billing.planautomatedupdatepayload.md
            section: []
          - title: firebase-functions/alerts/billing/planupdatepayload
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.billing.planupdatepayload.md
            section: []
      - title: firebase-functions/alerts/crashlytics
        path: >-
          /docs/functions/beta/reference/firebase-functions.alerts.crashlytics.md
        section:
          - title: firebase-functions/alerts/crashlytics/crashlyticsevent
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.crashlytics.crashlyticsevent.md
            section: []
          - title: firebase-functions/alerts/crashlytics/crashlyticsoptions
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.crashlytics.crashlyticsoptions.md
            section: []
          - title: firebase-functions/alerts/crashlytics/newanrissuepayload
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.crashlytics.newanrissuepayload.md
            section: []
          - title: firebase-functions/alerts/crashlytics/newfatalissuepayload
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.crashlytics.newfatalissuepayload.md
            section: []
          - title: firebase-functions/alerts/crashlytics/newnonfatalissuepayload
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.crashlytics.newnonfatalissuepayload.md
            section: []
          - title: firebase-functions/alerts/crashlytics/regressionalertpayload
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.crashlytics.regressionalertpayload.md
            section: []
          - title: firebase-functions/alerts/crashlytics/stabilitydigestpayload
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.crashlytics.stabilitydigestpayload.md
            section: []
          - title: firebase-functions/alerts/crashlytics/velocityalertpayload
            path: >-
              /docs/functions/beta/reference/firebase-functions.alerts.crashlytics.velocityalertpayload.md
            section: []
      - title: firebase-functions/alerts/firebasealertdata
        path: >-
          /docs/functions/beta/reference/firebase-functions.alerts.firebasealertdata.md
        section: []
      - title: firebase-functions/alerts/firebasealertoptions
        path: >-
          /docs/functions/beta/reference/firebase-functions.alerts.firebasealertoptions.md
        section: []
  - title: firebase-functions/cloudevent
    path: /docs/functions/beta/reference/firebase-functions.cloudevent.md
    section: []
  - title: firebase-functions/cloudfunction
    path: /docs/functions/beta/reference/firebase-functions.cloudfunction.md
    section: []
  - title: firebase-functions/eventarc
    path: /docs/functions/beta/reference/firebase-functions.eventarc.md
    section:
      - title: firebase-functions/eventarc/eventarctriggeroptions
        path: >-
          /docs/functions/beta/reference/firebase-functions.eventarc.eventarctriggeroptions.md
        section: []
  - title: firebase-functions/globaloptions
    path: /docs/functions/beta/reference/firebase-functions.globaloptions.md
    section: []
  - title: firebase-functions/https
    path: /docs/functions/beta/reference/firebase-functions.https.md
    section:
      - title: firebase-functions/https/callablefunction
        path: >-
          /docs/functions/beta/reference/firebase-functions.https.callablefunction.md
        section: []
      - title: firebase-functions/https/callablerequest
        path: >-
          /docs/functions/beta/reference/firebase-functions.https.callablerequest.md
        section: []
      - title: firebase-functions/https/httpserror
        path: /docs/functions/beta/reference/firebase-functions.https.httpserror.md
        section: []
      - title: firebase-functions/https/httpsoptions
        path: >-
          /docs/functions/beta/reference/firebase-functions.https.httpsoptions.md
        section: []
      - title: firebase-functions/https/request
        path: /docs/functions/beta/reference/firebase-functions.https.request.md
        section: []
  - title: firebase-functions/identity
    path: /docs/functions/beta/reference/firebase-functions.identity.md
    section:
      - title: firebase-functions/identity/blockingoptions
        path: >-
          /docs/functions/beta/reference/firebase-functions.identity.blockingoptions.md
        section: []
      - title: firebase-functions/identity/httpserror
        path: >-
          /docs/functions/beta/reference/firebase-functions.identity.httpserror.md
        section: []
  - title: firebase-functions/logger
    path: /docs/functions/beta/reference/firebase-functions.logger.md
    section:
      - title: firebase-functions/logger/logentry
        path: /docs/functions/beta/reference/firebase-functions.logger.logentry.md
        section: []
  - title: firebase-functions/params
    path: /docs/functions/beta/reference/firebase-functions.params.md
    section: []
  - title: firebase-functions/pubsub
    path: /docs/functions/beta/reference/firebase-functions.pubsub.md
    section:
      - title: firebase-functions/pubsub/message
        path: /docs/functions/beta/reference/firebase-functions.pubsub.message.md
        section: []
      - title: firebase-functions/pubsub/messagepublisheddata
        path: >-
          /docs/functions/beta/reference/firebase-functions.pubsub.messagepublisheddata.md
        section: []
      - title: firebase-functions/pubsub/pubsuboptions
        path: >-
          /docs/functions/beta/reference/firebase-functions.pubsub.pubsuboptions.md
        section: []
  - title: firebase-functions/storage
    path: /docs/functions/beta/reference/firebase-functions.storage.md
    section:
      - title: firebase-functions/storage/customerencryption
        path: >-
          /docs/functions/beta/reference/firebase-functions.storage.customerencryption.md
        section: []
      - title: firebase-functions/storage/storageevent
        path: >-
          /docs/functions/beta/reference/firebase-functions.storage.storageevent.md
        section: []
      - title: firebase-functions/storage/storageobjectdata
        path: >-
          /docs/functions/beta/reference/firebase-functions.storage.storageobjectdata.md
        section: []
      - title: firebase-functions/storage/storageoptions
        path: >-
          /docs/functions/beta/reference/firebase-functions.storage.storageoptions.md
        section: []
  - title: firebase-functions/tasks
    path: /docs/functions/beta/reference/firebase-functions.tasks.md
    section:
      - title: firebase-functions/tasks/authdata
        path: /docs/functions/beta/reference/firebase-functions.tasks.authdata.md
        section: []
      - title: firebase-functions/tasks/ratelimits
        path: /docs/functions/beta/reference/firebase-functions.tasks.ratelimits.md
        section: []
      - title: firebase-functions/tasks/request
        path: /docs/functions/beta/reference/firebase-functions.tasks.request.md
        section: []
      - title: firebase-functions/tasks/retrypolicy
        path: /docs/functions/beta/reference/firebase-functions.tasks.retrypolicy.md
        section: []
      - title: firebase-functions/tasks/taskqueuefunction
        path: >-
          /docs/functions/beta/reference/firebase-functions.tasks.taskqueuefunction.md
        section: []
      - title: firebase-functions/tasks/taskqueueoptions
        path: >-
          /docs/functions/beta/reference/firebase-functions.tasks.taskqueueoptions.md
        section: []
```

### CODE REVIEW

**Review by lahirumaramba — APPROVED** [review]
This looks pretty good to me! If Eric is good with this level of toc breakdown I think we are good.

**Review by egilmorez — COMMENTED** [review]
As I mentioned in our separate discussion, I'm concerned that pages generated but not included in this table of contents will be "orphaned," without displaying the table of contents. I think  Daniel is looking for a fix for that.

**Comment by taeold** [comment]
@egilmorez The current toc.yaml has more than 1 nested sections. Is this allowed?

File: package.json [hunk 1]
```diff
@@ -161,7 +161,8 @@
     "docgen:v1:gen": "api-documenter-fire markdown -i docgen/v1 -o docgen/v1/markdown && api-documenter-fire toc -i docgen/v1 -o docgen/v1/markdown/toc -p /docs/reference/functions",
     "docgen:v1": "npm run docgen:v1:extract && npm run docgen:v1:gen",
     "docgen:v2:extract": "api-extractor run -c docgen/api-extractor.v2.json --local",
-    "docgen:v2:gen": "api-documenter-fire markdown -i docgen/v2 -o docgen/v2/markdown && api-documenter-fire toc -i docgen/v2 -o docgen/v2/markdown/toc -p /docs/reference/functions/v2",
+    "docgen:v2:toc": "ts-node docgen/toc.ts --input docgen/v2/markdown --output docgen/v2/markdown/toc --path /docs/functions/beta/reference",
```
Comment (egilmorez) [inline package.json:164]: Daniel, I'm considering moving the .md files from

/docs/functions/beta/reference

to

/docs/reference/functions/v2

Is that just an update of this line, or would it be a PITA?

**Review by egilmorez — COMMENTED** [review]
This looks great, thanks!

One question on changing the path for the markdown files.

**Review by egilmorez — APPROVED** [review]
LG, thanks!

**Pull request merged**